airflow/models/baseoperator.py

```diff
@@ -1219,14 +1226,10 @@ def __deepcopy__(self, memo):
         return result
 
     def __getstate__(self):
-        state = dict(self.__dict__)
-        del state["_log"]
-
-        return state
+        return dict(self.__dict__)
```

Comment:
I think `_log` should still be deleted, and rebuilt in `__setstate__`.

Reply:
I admit the getstate/setstate part was not very clear to me.

I removed it because I assumed that a new `_log` object would be setup by the LoggingMixin anyway. But we have to keep it, is it because some Operators may customize their logging name ?
(I did not found such Operator, but I'd be happy to add a test case if someone point me out the purpose of this code).

In all cases, I'll put back the deletion and rebuilt :blush:

Reply:
Coming back on this one:

This chunk of code was added back in 2017 in #2838 (I cannot access the related Jira page), when the supported version were Python 2.7 to 3.4. There is a similar one in 4faaa13a0de9372d72f04ffb2f85bd7a01e0c25d to make the hive hook picklable.

Since Python 3.7, [loggers can be pickled](https://docs.python.org/3/whatsnew/3.7.html#logging). Thus, unless getstate / setstate are used for other things than pickling, I think this code can be removed.

Reply:
I think the main purpose is still for pickling although maybe something was added that also depended on it (don’t know). But since we have the needed information anyway, I would suggest still keeping this removal/rebuild code if only for performance.